cdap-client/src/main/java/co/cask/cdap/client/NamespaceClient.java

```diff
@@ -61,9 +62,18 @@ public NamespaceClient(ClientConfig config) {
    * @throws UnAuthorizedAccessTokenException if the request is not authorized successfully in the gateway server
    */
   public List<NamespaceMeta> list() throws IOException, UnAuthorizedAccessTokenException {
-    HttpResponse response = restClient.execute(HttpMethod.GET, config.resolveURL("namespaces"),
-                                               config.getAccessToken());
-    return ObjectResponse.fromJsonBody(response, new TypeToken<List<NamespaceMeta>>() { }).getResponseObject();
+    // TODO: remove the following apiVersion set/reset logic all APIs are migrated to v3
```

Comment:
@jwang47 not sure I understand completely, but I thought the logic of which API version to use be in the java client (which we control?). Allowing users to switch API versions might require us to take care of incompatibilities, etc?